interfaces/repo_test.go

```diff
@@ -43,26 +44,71 @@ var _ = Suite(&RepositorySuite{
 })
 
 func (s *RepositorySuite) SetUpTest(c *C) {
-	s.plug = &Plug{
-		Snap:      "provider",
-		Name:      "plug",
-		Interface: "interface",
-		Label:     "label",
-		Attrs:     map[string]interface{}{"attr": "value"},
-		Apps:      []string{"meta/hooks/plug"},
-	}
-	s.slot = &Slot{
-		Snap:      "consumer",
-		Name:      "slot",
-		Interface: "interface",
-		Label:     "label",
-		Attrs:     map[string]interface{}{"attr": "value"},
-		Apps:      []string{"app"},
-	}
+	// NOTE: the names provider/consumer are confusing. They will be fixed shortly.
+	provider, err := snap.InfoFromSnapYaml([]byte(`
+name: provider
+apps:
+    meta/hooks/plug:
+plugs:
+    plug:
+        interface: interface
+        label: label
+        attr: value
+`))
+	c.Assert(err, IsNil)
+	s.plug = &Plug{PlugInfo: provider.Plugs["plug"]}
+	consumer, err := snap.InfoFromSnapYaml([]byte(`
+name: consumer
+apps:
+    app:
+slots:
+    slot:
+        interface: interface
+        label: label
+        attr: value
+`))
+	c.Assert(err, IsNil)
+	s.slot = &Slot{SlotInfo: consumer.Slots["slot"]}
 	s.emptyRepo = NewRepository()
 	s.testRepo = NewRepository()
-	err := s.testRepo.AddInterface(s.iface)
+	err = s.testRepo.AddInterface(s.iface)
+	c.Assert(err, IsNil)
+}
+
+// plugFromYaml parses the given Yaml and returns Plug with the given name.
+func plugFromYaml(c *C, plugName string, yaml []byte) *Plug {
```

Comment:
I found a lot of the use cases of plugFromYaml and slotFromYaml to be much more verbose and confusing than the straightforward version of those. Many of the examples do not even have anything at all besides a plug/slot name. Can we please dial back on that and again have simpler code for these cases?